### PR TITLE
Included non-deps symbols in fuzzy matcher

### PR DESCRIPTION
**Pull request by scohen**
I noticed that the fuzzy matcher wouldn't match any unit tests. This is because their application is nil, as they're scripts. This change checks for nil applications, and if it finds one, only excludes the mapped entry if the path looks like it comes from a dependencies directory.

Note, the deps directory name is hard coded. I originally queried the project for the deps directory name, but this would include dependencies in the projects directory, which was quite annoying, hence the hard-coding.

### CODE REVIEW

File: apps/remote_control/lib/lexical/remote_control/search/fuzzy.ex [hunk 1]
```diff
@@ -359,8 +359,26 @@ defmodule Lexical.RemoteControl.Search.Fuzzy do
...
+      [" ", "deps", " "]
+      |> Path.join()
+      |> String.trim()
+
+    fn
+      mapped(application: nil, subtype: :definition, grouping_key: path) ->
+        # if we don't have an app name, just make sure we're not
+        # in what looks like a deps directory
+
+        not String.contains?(path, deps_part)
```
Comment (zachallaun) [inline apps/remote_control/lib/lexical/remote_control/search/fuzzy.ex:375]: What happens if our module has `Deps` in it? e.g. we're in `test/my_lib/parent/deps/child_test.exs`

Comment (scohen) [reply apps/remote_control/lib/lexical/remote_control/search/fuzzy.ex:375]: it won't show up.

Comment (scohen) [reply apps/remote_control/lib/lexical/remote_control/search/fuzzy.ex:375]: Technically, we'd need to interrogate each project's mix config to figure out what directory its deps reside in, `deps` is just the default, and I don't know how to go from app name -> mix project or from a directory to a mix project.

I can try to come up with a 80% heuristic though, that should be no worse than the above.

**Comment by scohen** [comment]
@zachallaun please take another look.

**Comment by scohen** [comment]
FYI, @zachallaun the previous approach wouldn't allow searches for `Lexical.Document` to succeed either.

File: apps/remote_control/lib/lexical/remote_control/search/fuzzy.ex [hunk 2]
```diff
@@ -336,31 +337,63 @@ defmodule Lexical.RemoteControl.Search.Fuzzy do
...
+        |> Path.join("deps")
+
+      if File.exists?(deps_dir) do
+        [deps_dir | deps_paths]
+      else
+        deps_paths
+      end
+    end)
+    |> Enum.reverse()
+    |> Enum.uniq()
```
Comment (zachallaun) [inline apps/remote_control/lib/lexical/remote_control/search/fuzzy.ex:393]: A slight simplification of this pipeline. Also, I think maybe the `uniq` can be excluded? I could be missing something obvious, but it seems like there shouldn't be duplicates.
```suggestion
    |> Enum.map(fn relative_mix_path ->
      relative_mix_path
      |> Path.absname()
      |> Path.dirname()
      |> Path.join("deps")
    end)
    |> Enum.filter(&File.exists?/1)
```

File: apps/remote_control/lib/lexical/remote_control/search/fuzzy.ex [hunk 3]
```diff
@@ -336,31 +337,63 @@ defmodule Lexical.RemoteControl.Search.Fuzzy do
...
+  end
+
+  defp deps_roots(%Project{mix_project?: true} = project) do
+    # Note: This function assumes that the deps directories for all
+    # found projects is `deps`. Projects may override this directory
+    # and lexical won't understand this. This was done because loading
+    # each sub-project is expensive and changes our global directory.
+
+    root_deps_path = Mix.Project.deps_path()
+    {root_deps_path, Project.mix_exs_path(project)}
```
Comment (zachallaun) [inline apps/remote_control/lib/lexical/remote_control/search/fuzzy.ex:374]: I think this is dead code or a leftover from debugging.
```suggestion
```

**Review by zachallaun — COMMENTED** [review]
Made a couple suggestions.

I do think this is a better approach than the previous iteration. One downside is that it finds the existing deps directories once when fuzzy is initialized, which means that any directories added later (from project restructuring/etc. maybe?) won't be ignored.

**Comment by scohen** [comment]
> One downside is that it finds the existing deps directories once when fuzzy is initialized

There's really no way to fix this. There is a lot of trouble with LSP file watching, several editors (emacs included) simply don't deliver `didChangeWatchedFiles` messages reliably, which is a shame.

**Review by zachallaun — APPROVED** [review]

**Pull request merged**